packages/block-editor/src/components/block-list/use-in-between-inserter.js

```diff
@@ -133,7 +135,8 @@ export function useInBetweenInserter() {
 				const clientId = element.id.slice( 'block-'.length );
 				if (
 					! clientId ||
-					__unstableIsWithinBlockOverlay( clientId )
+					__unstableIsWithinBlockOverlay( clientId ) ||
+					!! getParentSectionBlock( clientId )
```

Comment:
I'm still seeing the inserter in this PR, to me it seems to behave the same as trunk:

<video src="https://github.com/user-attachments/assets/9c05ee47-8857-4f01-af69-685ec86d5aa7"></video>

If I change this line then it seems to work as this PR describes:

```suggestion
					! getParentSectionBlock( clientId )
```

https://github.com/user-attachments/assets/7da44a47-610a-4d20-9f6e-a52cd6ee75b5

I'm not sure which is correct but I wanted to note I'm still seeing the inserter under this button.